src/viam/app/data/client.py

```diff
@@ -0,0 +1,159 @@
+    RemoveTagsFromBinaryDataByFileIDsRequest,
+    RemoveTagsFromBinaryDataByFilterRequest,
+    TagsByFilterRequest,
+    TagsByFilterResponse,
+    BoundingBoxLabelsByFilterRequest,
+    BoundingBoxLabelsByFilterResponse
+)
+from viam.proto.app.datasync import (
+    UploadMetadata,
+    SensorData,
+    FileData,
+)
+from viam.proto.app.datasync import DataSyncServiceStub
+
+
+class DataClient:
+    """gRPC client for uploading and retreiving data from app
+
+    Constructor is used by AppClient to instantiate relevant service stubs. Calls to DataClient methods should be made through AppClient.
+    """
+
+    def __init__(self, channel: Channel, metadata: str):
+        """
+        Create a data client that establishes a connection to app.
+
+        Args:
+            channel (Channel): an already-established connection to app.
+            metadata (str): the required authorization token to send requests to app.
+        """
+        self._metadata = metadata
+        self._data_client = DataServiceStub(channel)
+        self._data_sync_client = DataSyncServiceStub(channel)
+
+    _data_client: DataServiceStub
+    _data_sync_client: DataSyncServiceStub
+    _metadata: str
+
+    async def tabular_data_by_filter(
+        self,
+        filter: Optional[Filter],
+        dest: Optional[str]
+    ) -> List[Mapping[str, Any]]:
+        raise NotImplementedError()
+
+    async def binary_data_by_filter(
+        self,
+        data_request: Optional[Filter],
+        dest: Optional[str]
+    ) -> List[bytes]:
+        raise NotImplementedError()
+
+    async def binary_data_by_ids(self, file_ids: Optional[List[str]]) -> List[bytes]:
+        raise NotImplementedError()
+
+    async def delete_tabular_data_by_filter(self, filter: Optional[Filter]) -> None:
+        raise NotImplementedError()
+
+    async def delete_binary_data_by_filter(self, filter: Optional[Filter]) -> None:
+        raise NotImplementedError()
+
+    async def delete_binary_data_by_ids(self, file_ids: Optional[List[str]]) -> None:
+        raise NotImplementedError()
+
+    async def add_tags_to_binary_data_by_file_ids(self, file_ids: Optional[List[str]], tags: Optional[List[str]]) -> None:
+        """
+        Add tags to binary data using file ID's
+
+        Args:
+            file_ids (List[str]): List of file ID's specifying binary data to modify
+            tags (List[str]): List of tags to add to specified binary data
+        """
+        request = AddTagsToBinaryDataByFileIDsRequest(file_ids=file_ids, tags=tags)
+        await self._data_client.AddTagsToBinaryDataByFileIDs(request, metadata=self._metadata)
+
+    async def add_tags_to_binary_data_by_filter(self, tags: Optional[List[str]], filter: Optional[Filter] = Filter()) -> None:
+        """
+        Add tags to binary data using a filter
+
+        Args:
+            filter (Filter): Filter specifying binary data to modify
+            tags (List[str]): List of tags to add to specified binary data
```

Comment:
Same, should these be required?

Also, `Args` order should be in the same order as in the method signature